context.c

```diff
@@ -84,7 +86,13 @@ void *context_eval(engine_context *context, char *script) {
 
 	// Attempt to execute compiled string.
 	zval tmp;
-	_context_eval(op, &tmp);
+	_context_eval(op, &tmp, exit);
+
+	// Script called exit()
+	if (*exit != -1) {
```

Comment:
I think it may be possible to skip changing the signature of the version-specific functions (`_context_eval` and `_context_exec`) and just use `EG(exit_status)` here, since nothing else should be using the same context concurrently.